model/src/main/scala/aqua/model/ScriptModel.scala

```diff
@@ -1,24 +1,36 @@
 package aqua.model
 
 import aqua.model.func.{FuncCallable, FuncModel}
-import cats.Monoid
+import cats.{Monoid, Semigroup}
 import cats.data.Chain
 
 case class ScriptModel(
   funcs: Chain[FuncModel] = Chain.empty,
   services: Chain[ServiceModel] = Chain.empty,
-  types: Chain[TypeModel] = Chain.empty
+  types: Chain[TypeModel] = Chain.empty,
+  constants: Chain[ConstantModel] = Chain.empty
 ) extends Model {
 
+  case class Acc(
+    arrows: Map[String, FuncCallable],
+    values: Map[String, ValueModel]
+  )
+
   def resolveFunctions: Chain[FuncCallable] =
     funcs
```

Comment:
You fold over functions. That means that functions affect each other in textual order, but constants from the very bottom of a file affect all functions above: order is lost. I suppose ScriptModel should contain a single chain of parts so that the order of definitions is preserved.